torch/csrc/jit/script/module.h

```diff
@@ -53,13 +53,35 @@ struct Module;
 using ModuleLookup =
     std::function<std::shared_ptr<Module>(const std::vector<std::string>&)>;
 
+struct NamedIValue {
+  NamedIValue(std::string name, TypePtr type, IValue ivalue)
+      : name_(name),
+        type_(type),
+        ivalue_(torch::make_unique<IValue>(std::move(ivalue))) {}
+
+  Slot slot() const {
+    return Slot(ivalue_.get());
+  }
+  const std::string& name() const {
+    return name_;
+  }
+  const TypePtr& type() const {
+    return type_;
+  }
+
+ private:
+  const std::string name_;
+  const TypePtr type_;
+  std::unique_ptr<IValue> ivalue_;
+};
+
 struct Method {
   Method(
       Module* owner,
       std::string name,
       bool optimize,
       std::shared_ptr<Graph> graph,
-      std::vector<Slot> initial_members,
+      std::vector<const NamedIValue*> initial_members,
```

Comment:
NamedIValue's are objects stored continuously in a std::vector. Appending to that vector invalidates the pointers to that vector. So these lists of NamedIValue* are unsafe. If this is true, then this is a major bug and we need to revert.